.circleci/config.yml

```diff
@@ -162,14 +162,27 @@ jobs:
           name: Build Docker artifact
           command: docker build --pull -t "cosmwasm/wasmd:${CIRCLE_SHA1}" .
       - run:
-          name: Push application Docker image to docker hub
+          name: Ensure libwasmvm version is correct
           command: |
-            if [ "${CIRCLE_BRANCH}" = "master" ]; then
-              docker tag "cosmwasm/wasmd:${CIRCLE_SHA1}" cosmwasm/wasmd:latest
-              docker login --password-stdin -u "$DOCKER_USER" \<<<"$DOCKER_PASS"
-              docker push cosmwasm/wasmd:latest
-              docker logout
+            IN_DOCKER=$(docker run --rm "cosmwasm/wasmd:${CIRCLE_SHA1}" /usr/bin/wasmd query wasm libwasmvm-version)
+            echo "Runtime libwasmvm-version in docker: $IN_DOCKER"
+            IN_GOMOD=$(go list -m github.com/CosmWasm/wasmvm | cut -d" " -f2 | cut -d"v" -f2)
+            echo "wasmvm version in go.mod: $IN_GOMOD"
+            if [[ "$IN_DOCKER" != "$IN_GOMOD" ]]; then
+              echo "Mismatch of wasmvm versions detected"
+              exit 1
             fi
+      - when:
+          condition:
+            equal: [ main, << pipeline.git.branch >> ]
```

Comment:
This pushes to docker hub on main brach builds only, right?

Reply:
Right, that's the idea. Not 100 % tested but it is coming from the docs. At least in this PR the `when` step is skipped because the condition is not met. Let's see what is happening on main.